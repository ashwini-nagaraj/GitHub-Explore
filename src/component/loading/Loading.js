import React from "react";

import loading from "../../loading.gif";
import "./Loading.css";

const Loading = () => (
  <div data-testid="loading"  className="loading">
    <img data-testid="image" src={loading} alt="loading" />
  </div>
);

export default Loading;