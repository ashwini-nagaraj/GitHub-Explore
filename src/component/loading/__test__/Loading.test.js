import React from "react";
import ReactDOM from "react-dom";
import Loading from "../Loading";
import { render , cleanup} from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup)

it("renders without crashig", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Loading></Loading>, div);
});

it("renders loading correctly", () => {
  const { getByTestId } = render(
    <Loading>
    </Loading>
  );
  const div = getByTestId("loading");
  const image = getByTestId("image");
  expect(div).toContainElement(image);
});


it("matches snapshot",()=>{
   const tree=  renderer.create(<Loading></Loading>).toJSON();
    expect(tree).toMatchSnapshot();
})