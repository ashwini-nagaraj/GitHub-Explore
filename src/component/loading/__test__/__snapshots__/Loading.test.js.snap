// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="loading"
  data-testid="loading"
>
  <img
    alt="loading"
    data-testid="image"
    src="loading.gif"
  />
</div>
`;
